src/services/infra/InfraService.ts

```diff
@@ -91,4 +105,174 @@ export default class InfraService {
       throw err
     }
   }
+
+  parseDNSRecords = (record?: string) => {
+    if (!record) {
+      return undefined
+    }
+
+    // Note: the records would have the shape of 'blah.gov.sg CNAME blah.validaations.aws'
+    const recordsInfo = record.split(" ")
```

Comment:
> aim to parse and preserve it in our types 
May I get your help to give an example? are you suggesting to check if `recordsInfo[0]` and `recordsInfo[2]` are valid urls?

Reply:
hmm because this invariant seems important, we want to ensure that 1. there's some semantic information being attached here (ie, create a `type`) and 2. this method is the only way for us to construct said type and hence, allows us to maintain said invariants.

what this would entail here is perhaps 1. validating url (we use `joi` so [this](https://joi.dev/api/?v=17.7.0#stringdomainoptions) might be useful) and 2. that the `type` (`CNAME` | `A` etc) is valid.

because this method is q lenient atm, consumers might call it on something like `"invalid string"`, and run into errors down the line

Reply:
438f947